lib/spack/spack/installer.py

```diff
@@ -2146,6 +2167,7 @@ def install(self) -> None:
                     # or uninstalled -- on the next pass.
                     self.installed.remove(pkg_id)
                     self._requeue_task(task, install_status)
+                t.stop("task")
```

Comment:
seems like you really want to use a context manager here instead of repeating t.stop on every exit point

Reply:
nope and neither is there go like defer